### ISSUE TITLE
DSN transmitting frequency ramp tables overlap

### ISSUE THREAD

**Issue by valeriof7** [tudat-team/tudatpy#484]
In PR tudat-team/tudat#268, the following code block in `include/tudat/astro/ground_stations/transmittingFrequencies.h` and relative unit test has been commented out, but needs to be addressed:
"""
// If there are discontinuities
            if ( startTimes_.at( i ) != endTimes_.at( i - 1 ) )
            {
                // If the start and end times are inconsistent throw error
                if ( endTimes_.at( i - 1 ) > startTimes_.at( i ) )
                {
                    throw std::runtime_error(
                            "Error when creating piecewise linear frequency interpolator: inconsistency between ramp end "
                            "time (" + std::to_string( double( endTimes_.at( i - 1 ) ) ) + ") and start time of the following ramp (" +
                            std::to_string( double( startTimes_.at( i ) ) ) + "); the end is smaller than the start time." );
                }
                // If there are gaps in the data save that information
                else
                {
                    invalidTimeBlocksStartTimes_.push_back( endTimes_.at( i - 1 ) );
                    invalidTimeBlocksEndTimes_.push_back( startTimes_.at( i ) );
                }
"""

In particular, we need to properly implement the logic to deal with overlaps in the DSN transmit frequency ramp table. Possible solutions suggested by @DominicDirkx:
- check if the tables are only slightly different (same frequencies possibly referenced to a different date)
- select one of the tables and reset the time range of the other one
- give a warning
- give an exception

**Comment by DominicDirkx** [comment]
This PR: https://github.com/tudat-team/tudatpy/pull/563 made some modifications to the loading of transmission frequencies from IFMS files that are tangentially related to this.

I think there are a couple of related issues here:

1 What to do if a frequency is requested outside of the bounds where data is available?
2 What to do if a frequency is requested at a point where multiple ramp tables are available?

I think it's important to not hard-code anything, but let all values we use be user-definable (but, with nominal values if the user does not provide anything)

For (1), I would propose we simply extend the frequency data (extrapolate it in time beyond the formally valid region) for a given number of seconds (default 10 s?). This will be useful in case the the frequency is queried *just* before/after the definition of the ramp tables.  The more complicated question is what to do outside these bounds

* Throwing an error: may make the code crash when in case of (for instance) outliers
* Giving a warning: can spam the terminal with many warnings
* Just extrapolating further in time: can hide the fact that data is missing

We can support each of these options, I would be in favour of using the warning as default, while providing a single function that turns these warnings off for *all* ground stations (in addition to more fine-grained control)

For (2), if the overlap between ramps is small (less than 10 s?), we can just pick one. Most likely it won't matter much, and if it does the Doppler calculation may be wrong and a small number of data points will get flagged as outliers. Here, I think the bigger issue is what if ramp tables with significant overlap are added? My proposal for a default would be (again) printing a warning, but then using the later definition of ramp tables, since here this is *probably* not something a user wants to do.

**Comment by valeriof7** [comment]
When I wrote the frequency processor for the DSN ramps, I used the [following logic](https://github.com/valeriof7/tudatpy/blob/3d0d4ce5efc61aaaaccc66d6877428885772785a/src/tudatpy/data/processTrk234/converters/ramp.py#L38), which, in theory, should bypass that check because it is performed in the Python layer. I'm not very familiar with IFMS files, but DSN files do not provide an end time. Basically, they only provide a time, frequency, and rate. If applicable to IFMS, the logic implemented in `ramp.py` would be the one I'd go for.
To summarize:
1. It is okay to extrapolate, as was done for DSN stations.
2. I would issue a warning and set the flag for the measurements to "Ignore" so that they won't be used in filtering.
